crates/red_knot_wasm/src/lib.rs

```diff
@@ -223,40 +224,43 @@ impl FileHandle {
 #[wasm_bindgen]
 pub struct Diagnostic {
     #[wasm_bindgen(readonly)]
-    inner: Box<dyn OldDiagnosticTrait>,
+    inner: RefCell<diagnostic::Diagnostic>,
```

Comment:
The `RefCell` might no longer be necessary, now that `Diagnostic::print` takes `&self`